File: house-robber/house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        evenm=0
        oddm=0
        if len(nums)==1:
            return nums[0]
        elif len(nums)==2:
            return max(nums[0],nums[1])
            
        for i in range(len(nums)):
            if i%2==0:
                evenm+=nums[i]
                evenm=max(evenm,oddm)
            else:
                oddm+=nums[i]
                oddm=max(oddm,evenm)
        return max(oddm,evenm)
                
        
                    
            
            

File: house-robber/house-robber.java
class Solution {
    public int rob(int[] nums) {
        int n= nums.length;
        int[] dp = new int[nums.length+2];
        int j=2;
        for(int i=n-1;i>=0;i--){
            dp[j] = Math.max(dp[j-2] + nums[i],dp[j-1]);
            j++;
        }
        return dp[n+1];
    }
}